src/model/tree/DataVirtNodeProvider.ts

```diff
@@ -97,11 +96,11 @@ export class DataVirtNodeProvider implements vscode.TreeDataProvider<vscode.Tree
 		return new Promise<void>( async (resolve, reject) => {
 			this.resetList();
 
-			fs.readdirSync(this.workspace).forEach( (file) => {
-				if (file.toLowerCase().endsWith('.yaml')) {
-					this.processFile(this.workspace, file);
+			for (const [name, type] of await vscode.workspace.fs.readDirectory(vscode.Uri.file(this.workspace))) {
```

Comment:
what do you think to extract the await statements as a const? it will avoid several statements to be imbricated